core/ledger/ledgerstorage/store.go

```diff
@@ -143,7 +143,6 @@ func (s *Store) CommitWithPvtData(blockAndPvtdata *ledger.BlockAndPvtData) error
 	}
 
 	if err := s.AddBlock(blockAndPvtdata.Block); err != nil {
-		s.pvtdataStore.Rollback()
```

Comment:
Given that we are removing the _rollback()_, having _commit()_ may not make sense now. As already there is a huge difference in code between v1.4 and v2.0, having these additional differences might create more confusion in the future. Why not pick this PR https://github.com/hyperledger/fabric/commit/b33e5aaea8ebb618ff9881d7f6ac5edd4709625f#diff-596a7deb3a7e702099aac438aede9984 to v1.4 to reduce the code differences? If not, are we trying to minimize the changes made to v1.4?

Reply:
generally, we try to minimize changes where possible and sometimes this means we can't cherry pick a change but have to "re-implement" part of it.

Reply:
Yes, minimizing the changes was the main motive. In addition, on the code difference part, 1.4 and 2.0 looks significantly different now in many aspects and I have accepted the mode of maintaining two code bases and try to make changes in 1.4 branch such that it looks closer to earlier releases of 1.4.x as opposed to close to 2.0

Reply:
Thanks, @mastersingh24 and @manish-sethi for the clarification. The changes look good to me.